card/mod/01_core/chunk/query_reference.rb

```diff
@@ -0,0 +1,65 @@
+# -*- encoding : utf-8 -*-
+
+require_dependency File.expand_path( '../reference', __FILE__ )
+
+module Card::Chunk
+
+  # This should find +Alfred+ in expressions like
+  # 1) {"name":"Alfred"}
+  # 2a) {"name":["in","Alfred"]}
+  # 3a) {"plus_right":["Alfred"]}
+  # but not in
+  # 2b) "content":"foo", "Alfred":"bar"
+  # 3b) {"name":["Alfred", "Toni"]}
+  # It's not possible to distinguish between 2a) and 2b) or 3a) and 3b) with a simple regex
+  # hence we use a too general regex and check for query keywords after the match
+  # which of course means that we don't find references with query keywords as name
+  class QueryReference < Reference
```

Comment:
I suspect we may eventually want to move the JSON parsing into chunk code, especially once we have caching for parsed content.